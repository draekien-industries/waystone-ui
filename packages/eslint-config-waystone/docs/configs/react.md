# eslint-config-waystone - react

## Usage

```js
// .eslintrc.js
module.exports = {
  extends: ['waystone/react'],
};
```

## Env

- browser
- es6
- shared-node-browser

## Extends

- airbnb
- turbo
- eslint:recommended
- import/recommended
- import/react
- import/typescript
- @typescript-eslint/eslint-recommended
- @typescript-eslint/recommended
- react/recommended
- react/jsx-runtime
- waystone/recommended
- jsx-a11y/recomended
- testing-library/recommended
- jest/recommended
- prettier
