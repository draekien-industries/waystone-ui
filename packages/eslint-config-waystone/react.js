module.exports = {
  env: {
    browser: true,
    es6: true,
    'shared-node-browser': true,
  },
  extends: [
    'airbnb',
    'turbo',
    'eslint:recommended',
    'plugin:import/recommended',
    'plugin:import/react',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:waystone/recommended',
    'plugin:jsx-a11y/recommended',
    'prettier',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 'latest',
    sourceType: 'module',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
  rules: {
    'no-plusplus': ['error', { allowForLoopAfterthoughts: true }],
    '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
    '@typescript-eslint/consistent-type-imports': [
      'error',
      {
        prefer: 'type-imports',
        fixStyle: 'separate-type-imports',
      },
    ],
    '@typescript-eslint/no-unused-vars': [
      'error',
      { varsIgnorePattern: '^_*' },
    ],
    'react/function-component-definition': [
      'warn',
      {
        namedComponents: 'arrow-function',
        unnamedComponents: 'arrow-function',
      },
    ],
    'react/jsx-filename-extension': ['warn', { extensions: ['.tsx'] }],
    'react/jsx-props-no-spreading': 'off',
    'react/no-unknown-property': ['error', { ignore: ['sx'] }],
    'react/require-default-props': 'off',
    'import/extensions': [
      'error',
      'ignorePackages',
      { ts: 'never', tsx: 'never', js: 'never', jsx: 'never' },
    ],
    'import/prefer-default-export': 'off',
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'no-use-before-define': [
      'error',
      {
        functions: false,
        classes: true,
        variables: true,
        allowNamedExports: false,
      },
    ],
  },
  overrides: [
    {
      env: { jest: true },
      files: [
        '**/__tests__/**/*.[jt]s?(x)',
        '**/__fixtures__/**/*.[jt]s?(x)',
        '**/?(*.)+(spec|test).[jt]s?(x)',
      ],
      extends: [
        'airbnb',
        'turbo',
        'eslint:recommended',
        'plugin:import/recommended',
        'plugin:import/typescript',
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:react/recommended',
        'plugin:react/jsx-runtime',
        'plugin:react-hooks/recommended',
        'plugin:waystone/recommended',
        'plugin:jsx-a11y/recommended',
        'plugin:testing-library/react',
        'plugin:jest/recommended',
        'prettier',
      ],
      rules: {
        'import/no-extraneous-dependencies': [
          'off',
          { devDependencies: ['**/?(*.)+(spec|test).[jt]s?(x)'] },
        ],
        'react/jsx-filename-extension': ['warn', { extensions: ['.tsx'] }],
        'import/extensions': [
          'error',
          'ignorePackages',
          { ts: 'never', tsx: 'never', js: 'never', jsx: 'never' },
        ],
      },
    },
  ],
  ignorePatterns: [
    '**/*.js',
    '**/*.json',
    '**/*.d.ts',
    '**/*.g.*',
    '**/*.generated.*',
    'node_modules',
    'public',
    'styles',
    '.next',
    'coverage',
    'dist',
    '.turbo',
  ],
};
