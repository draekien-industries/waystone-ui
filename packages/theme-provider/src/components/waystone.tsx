import type { PropsWithChildren } from 'react';
import { type Theme, ThemeUIProvider } from 'theme-ui';
import { waystoneTheme } from '@waystone/theme';
import { mergeObjects } from './waystone.utils';

export type WaystoneProps = {
  theme?: Partial<Theme>;
};

export const Waystone = ({
  children,
  theme,
}: PropsWithChildren<WaystoneProps>) => (
  <ThemeUIProvider theme={mergeObjects(waystoneTheme, theme)}>
    {children}
  </ThemeUIProvider>
);
