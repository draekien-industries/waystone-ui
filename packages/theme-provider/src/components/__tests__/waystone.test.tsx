import { render } from '@testing-library/react';
import { Waystone } from '../waystone';

describe('The Waystone component', () => {
  it('Matches snapshot', () => {
    const { container } = render(<Waystone>test</Waystone>);

    expect(container).toMatchSnapshot();
  });
});
