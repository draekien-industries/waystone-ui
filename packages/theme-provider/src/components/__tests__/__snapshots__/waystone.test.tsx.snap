// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Waystone component Matches snapshot 1`] = `
<div>
  test
</div>
`;
